close #1274: Don't hide popover that has child popover on click in child

pdanpdan opened the pull request:
- also refocus QAutocomplete input element after click in option (so the behavior is the same as on enter)
- consider an anchorEl for popover if it's q-if-inner
- don't use capture for __bodyHide, so we can stop propagation

<!--
Please make sure to read the Pull Request Guidelines:
https://github.com/quasar-framework/quasar/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [X] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [X] It's submitted to the `dev` branch and _not_ the `master` branch
- [X] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [X] It's been tested with all Quasar themes
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar-framework.org/tree/master/source) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
close #1274

alexdee2007 commented on src/components/autocomplete/QAutocomplete.js:
```diff
@@ -148,6 +149,10 @@ export default {
     setValue (result) {
+      this.enterKey = true
+      if (this.inputEl && this.__input && !this.__input.hasFocus()) {
+        this.inputEl.focus()
```
Hmm... Not sure if it useful. When you working with mouse or tuch it does not matter...
It only matters when you work with the keyboard... And there everything is ok...
Besides, double assignment **this.enterKey = true** when Enter press

pdanpdan replied:
Fixed duplicate this.enterKey = true.
It matters to have it set because it blocks reopening the popover with the list when the value is set in parent.

alexdee2007 replied:
>when the value is set in parent.

Can you explain more detail? I cannot reproduce it... Just wondering is #1282 not fixes this problem?

pdanpdan replied:
The idea is like this:
- parent component of QAutocomplete has a shadow property that keeps the value as `val` - this is updated according to debouncing and some delays for processing
- if we listen for input from the parent we're going to get in `val` the value before the last key press 
- if we listen for changes in `val` we get the correct value every time, but that value we are sometimes updating from the autocomplete, so we want to skip showing the options in that case

alexdee2007 replied:
Well, its clear ) But we can use target.value from input event? Then there will be a lot of redundant code and checks? I'll play around with this)
I still think it is improperly to use a watcher for autocomplete...

alexdee2007 commented on src/components/popover/QPopover.js:
```diff
@@ -74,7 +74,7 @@ export default {
       this.anchorEl = this.$el.parentNode
       this.anchorEl.removeChild(this.$el)
-      if (this.anchorEl.classList.contains('q-btn-inner')) {
+      if (this.anchorEl.classList.contains('q-btn-inner') || this.anchorEl.classList.contains('q-if-inner')) {
```
This is really a very useful addition!

pdanpdan committed:
close #1274: Don't hide popover that has child popover on click in child

- also refocus QAutocomplete input element after click in option (so the behavior is the same as on enter)
- consider an anchorEl for popover if it's q-if-inner
- don't use capture for __bodyHide, so we can stop propagation
